libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/ReplaceDialog.cs

```diff
@@ -78,5 +79,14 @@ public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc,
             // Replace dialog with bound options passed in as the options
             return await dc.Parent.ReplaceDialogAsync(dialog.Id, options: boundOptions, cancellationToken: cancellationToken).ConfigureAwait(false);
         }
+
+        /// <summary>
+        /// Enumerate child dialog dependencies required to be exposed for parent containers dialogset.
+        /// </summary>
+        /// <returns>dialog enumeration.</returns>
+        public virtual IEnumerable<Dialog> GetExternalDependencies()
```

Comment:
Once we move `IAdaptiveDialogDependencies` to the `Microsoft.Bot.Builder.Dialogs.Adaptive` assembly, I would love to make this internal so that it's not publicly visible, but unfortunately internal methods cannot implement internal interfaces in C#, which means that we cannot get away from making this public.